crates/c-api/src/types/global.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{wasm_externtype_t, wasm_valtype_t, CExternType};
+use once_cell::unsync::OnceCell;
+use wasmtime::GlobalType;
+
+pub type wasm_mutability_t = u8;
+
+#[repr(transparent)]
+#[derive(Clone)]
+pub struct wasm_globaltype_t {
+    ext: wasm_externtype_t,
+}
+
+wasmtime_c_api_macros::declare_ty!(wasm_globaltype_t);
+
+#[derive(Clone)]
+pub(crate) struct CGlobalType {
+    pub(crate) ty: GlobalType,
+    content_cache: OnceCell<wasm_valtype_t>,
+}
+
+impl wasm_globaltype_t {
+    pub(crate) fn new(ty: GlobalType) -> wasm_globaltype_t {
+        wasm_globaltype_t {
+            ext: wasm_externtype_t::new(ty.into()),
+        }
+    }
+
+    pub(crate) fn try_from(e: &wasm_externtype_t) -> Option<&wasm_globaltype_t> {
+        match &e.which {
+            CExternType::Global(_) => Some(unsafe { &*(e as *const _ as *const _) }),
+            _ => None,
+        }
+    }
+
+    pub(crate) fn ty(&self) -> &CGlobalType {
+        match &self.ext.which {
+            CExternType::Global(f) => &f,
+            _ => unsafe { std::hint::unreachable_unchecked() },
+        }
+    }
+}
+
+impl CGlobalType {
+    pub(crate) fn new(ty: GlobalType) -> CGlobalType {
+        CGlobalType {
+            ty,
+            content_cache: OnceCell::new(),
+        }
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn wasm_globaltype_new(
+    ty: Box<wasm_valtype_t>,
+    mutability: wasm_mutability_t,
+) -> Box<wasm_globaltype_t> {
+    use wasmtime::Mutability::*;
+    let mutability = match mutability {
+        0 => Const,
```

Comment:
I doubt there would ever be more than 0 and 1 as values, but do we want symbolic representation of the C enum rather than the hardcoded values here and in `wasm_globaltype_mutability`?